src/main/java/net/imglib2/algorithm/gradient/PartialDerivative.java

```diff
@@ -234,4 +234,142 @@
 			}
 		}
 	}
+	
+	/**
+	 * Compute the backward difference of source in a particular dimension.
```

Comment:
Please at some JavaDoc math that specifies exactly what is being computed, like
https://en.wikipedia.org/wiki/Finite_difference#Forward,_backward,_and_central_differences

Reply:
Using the LoopBuilder implementation by @maarzt makes the code much more readable. If necessary I can add more java doc.

Reply:
Having the math in the doc would be very helpful, either in code
```
d_f( x ) = f( x + h ) - f( x )
```
or tex
[![](https://wikimedia.org/api/rest_v1/media/math/render/svg/9b4b1b463ac6a532d0089dad4ef7f7ec1a91acb9)](https://en.wikipedia.org/wiki/Finite_difference#Forward,_backward,_and_central_differences)
(examples for forward difference).

Reply:
The function `f` in the equation `d_f( x ) = f( x + h ) - f( x )` has an one dimensional domain. If we want to describe what `gradientForwardDifference` actually computes on 2d, 3d or Nd images, we would need a more complicated equation.
I think the resulting math will be rather confusing. @hanslovsky Please proof me wrong!